doc/source/whatsnew/v0.25.0.rst

```diff
@@ -641,6 +641,7 @@ Performance improvements
 - Restored performance of :meth:`DatetimeIndex.__iter__` by re-enabling specialized code path (:issue:`26702`)
 - Improved performance when building :class:`MultiIndex` with at least one :class:`CategoricalIndex` level (:issue:`22044`)
 - Improved performance by removing the need for a garbage collect when checking for ``SettingWithCopyWarning`` (:issue:`27031`)
+- RangeIndex now performs standard lookup without instantiating an actual hashtable, hence saving memory (:issue:`16685`)
```

Comment:
i think we have another perf issue for RangeIndex
can u move it next to it; also can u use the ‘:class:RangeIndex’